### ISSUE TITLE
Transpose multidimensional array

### ISSUE THREAD

**Issue by jdmcclain47** [dask/dask#1630]
Hi, I was wondering if there was any way to transpose a multidimensional array.  Unfortunately I have to work with tensors with fairly long dimensions and transposing is a necessity.

```
import numpy as np
from distributed import Client
import dask.dataframe as dd
import dask.array as da

nocc = 20
nvir = 4
nkpts = 8
nkpts_blk = 2
oovv   = da.ones((nkpts,nkpts,nkpts,nocc,nocc,nvir,nvir),chunks=(nkpts_blk,nkpts_blk,nkpts_blk,nocc,nocc,nvir,nvir))
```

In numpy I can do something like 
`oovvt = oovv.transpose(0,1,2,3,4,6,5)`
but this doesn't seem to work.  It's across an axis that isn't "chunked" per-say, so it wouldn't be that terrible to do.

**Comment by mrocklin** [comment]
Transpose is implemented.  However it looks like it expects a list rather than splatted out entries.  

```python
In [1]: import dask.array as da

In [2]: nocc = 20
   ...: nvir = 4
   ...: nkpts = 8
   ...: nkpts_blk = 2
   ...: oovv   = da.ones((nkpts,nkpts,nkpts,nocc,nocc,nvir,nvir),chunks=(nkpts_b
   ...: lk,nkpts_blk,nkpts_blk,nocc,nocc,nvir,nvir))
   ...: 

In [3]: oovv
Out[3]: dask.array<wrapped..., shape=(8, 8, 8, 20, 20, 4, 4), dtype=float64, chunksize=(2, 2, 2, 20, 20, 4, 4)>

In [4]: oovv.transpose(axes=[0, 1, 2, 3, 4, 6, 5])
Out[4]: dask.array<transpo..., shape=(8, 8, 8, 20, 20, 4, 4), dtype=float64, chunksize=(2, 2, 2, 20, 20, 4, 4)>
```

This API mismatch looks like a bug to me.

**Comment by shoyer** [comment]
NumPy accepts both forms... sigh... http://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.transpose.html

**Comment by jcrist** [comment]
Fixed in #1632.
